app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,31 @@
+class InvitationsController < ApplicationController
+  before_action :require_user
+
+  def new
+    @invitation = Invitation.new
+  end
+
+  def create
+    @invitation = Invitation.new(invitation_params)
+    if @invitation.save
+      send_invitation_email
+      redirect_to new_invitation_path
+    else
+      flash[:danger] = "Invalid input"
+      render :new
+    end
+  end
+
+  def send_invitation_email
+    AppMailer.send_invitation(@invitation).deliver
+    flash[:success] = "Invitation sent!"
```

Comment:
I'm not sure I think this belongs here. The method name doesn't mention the flash message, and this is tacking on a second responsibility. If it was my code, I think I would leave the flash assignment in the control flow blow.